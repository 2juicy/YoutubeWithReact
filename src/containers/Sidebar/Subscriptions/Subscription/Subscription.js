import React from "react";
import { Icon, Image, Menu } from "semantic-ui-react";
import "./Subscription.scss";

export default function Subscription({ broadcasting, newVids, label }) {
  return (
    <Menu.Item className="sidebar-item">
      <div className="subscription ">
        <div className="flex-text">
          <Image avatar src="https://via.placeholder.com/28x28" />
          <span>{label}</span>
        </div>
        {broadcasting ? (
          <Icon name="signal" />
        ) : newVids ? (
          <span className="new-vids-count">{newVids}</span>
        ) : null}
      </div>
    </Menu.Item>
  );
}
