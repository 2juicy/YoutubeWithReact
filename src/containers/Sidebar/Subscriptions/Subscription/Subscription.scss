.subscription {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
  .flex-text {
    display: flex;
    align-items: center;
    img {
      margin-right: 20px !important;
    }
    span {
      line-height: 28px;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    }
  }

  i.icon {
    color: red;
  }

  .new-vids-count {
    color: gray;
  }
}
