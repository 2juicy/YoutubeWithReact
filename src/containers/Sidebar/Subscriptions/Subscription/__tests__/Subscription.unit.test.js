import React from "react";
import { shallow } from "enzyme";
import Subscription from "../Subscription";

describe("Subscription", () => {
  test("renders", () => {
    const wrapper = shallow(<Subscription />);
    expect(wrapper).toMatchSnapshot();
  });
});
