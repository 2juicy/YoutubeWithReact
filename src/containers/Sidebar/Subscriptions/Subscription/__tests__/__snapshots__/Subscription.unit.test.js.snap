// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscription renders 1`] = `
<MenuItem
  className="sidebar-item"
>
  <div
    className="subscription "
  >
    <div
      className="flex-text"
    >
      <Image
        as="img"
        avatar={true}
        src="https://via.placeholder.com/28x28"
        ui={true}
      />
      <span />
    </div>
  </div>
</MenuItem>
`;
